libsolidity/interface/CompilerStack.cpp

```diff
@@ -597,6 +597,15 @@ bool CompilerStack::analyze()
 			if (m_modelCheckerSettings.engine.any())
 				m_modelCheckerSettings.solvers = ModelChecker::checkRequestedSolvers(m_modelCheckerSettings.solvers, m_errorReporter);
 
+			// Tiny automatic optimization for Eldarica if the user
+			// has not forcefully chosen `divModNoSlacks = false` which
+			// is the default for Spacer but worse for Eldarica.
+			if (
+				m_modelCheckerSettings.solvers.eld &&
+				!m_modelCheckerSettings.divModNoSlacks.has_value()
+			)
+				m_modelCheckerSettings.divModNoSlacks = true;
```

Comment:
I'd rather have this replaced with a warning than changing the settings during runtime, but I guess if you guys (SMT people) find this to be more convenient, I can live with it.